### DISCUSSION TITLE
Unique select in the ModelFormSet

### DISCUSSION THREAD

**Discussion by rez0n** [codingjoe/django-select2#152] [Q&A]
Hi, I have a dialog where user should pick other users to belong to some group with additional parameters.
I implemented ModelFormSet to render multiple rows with user selection using django-select2, there should be limitation to avoid selecting the same user twice, I implemented customized clean method for the formset and it works fine, but I need to improve this to make unable selecting the user which is already selected in another select2 field.
Any recommendations to approach for this?
Much thanks!

**Comment by codingjoe** [comment]
Hi @rez0n,

Thanks for reaching out. I believe you're looking for [dependent_fields](https://django-select2.readthedocs.io/en/latest/extra.html#interdependent-select2). However, in your use-case you will need to overwrite `filter_queryset` of your dependee to implement the exclusion properly.

I hope those pointers help. If you need an example, let me know.

Cheers,
Joe

**Reply by rez0n** [reply]
Hi @codingjoe 
I wasn't sure is field could be dependent on himself (actually another field but a full duplicate of itself)
I going to check this and then back with the result. Overwriting of the `filter_queryset` is not a problem, but I am not sure how it will work on the client (js) side.

Thanks for your advise.

**Comment by rez0n** [comment]
Actually I managed to do this in next way

I have overwritten django_select2.js for a particular view in a way to create list of already selected id's for a field

```
var dependentFields = $element.data('select2-dependent-fields')
if (dependentFields) {
    dependentFields = dependentFields.trim().split(/\s+/)
    $.each(dependentFields, function (i, dependentField) {
        let ids = [];
        $('[name$=' + dependentField + ']').each(function() {
            let personId = $(this).val();
            if (personId) {
                ids.push(personId);
            }
        });
        result[dependentField] = ids
    })
}
```
This way `s2forms.ModelSelect2Widget` receive this list in `**dependent_fields` so I can manipulate them
```
class PersonDependentPickWidget(s2forms.ModelSelect2Widget):
    dependent_fields = {
        'person': 'any_model_related_name_field',
    }

    ....

    def filter_queryset(self, request, term, queryset, **dependent_fields):
        queryset = super().filter_queryset(request, term, queryset)
        select = Q()

        if dependent_fields:
            selected_person_ids = dependent_fields.get('any_model_related_name_field__in', None)
            select &= ~Q(pk__in=selected_person_ids)

        queryset = queryset.filter(select)
        return queryset
```

I'd already had a [similar task](https://github.com/codingjoe/django-select2/issues/136) to overwrite django_select2.js to get the dependent field in FormSet, this one is a bit different because it may return list of selected values instead single value. 
Would be cool to update package to make JS universal for all cases, maybe worth to add a special class (like `django-select2-dependent-{{field_name}}`) to make it able to select it by class in formsets
